lib/classification_dump_cache.rb

```diff
@@ -15,10 +16,20 @@ def reset_subject_workflow_statuses(subject_workflow_statuses)
     @subject_workflow_statuses = subject_workflow_statuses.group_by(&:subject_id)
   end
 
+  def reset_classification_subjects(classification_subjects)
+    @classification_to_subjects = classification_subjects.map do |cs|
```

Comment:
This won't work for classifications with more than one subject, you'll need to group `classification_subjects` by `classification_id`, which will also make this entire body simpler:

```ruby
@classification_to_subjects = classification_subjects.group_by {|classification_id, _| classification_id }
```

Of course, if you go the model route that'll need to be different than my example.